spec/controllers/v1.0/requests_controller_spec.rb

```diff
@@ -166,17 +255,64 @@
     let(:item) { { 'disk' => '100GB' } }
     let(:valid_attributes) { { :requester_name => '1234', :name => 'Visit Narnia', :content => item, :description => 'desc' } }
 
-    context 'when request attributes are valid' do
+    context 'admin role when request attributes are valid' do
+      let(:access_obj) { instance_double(RBAC::Access, :accessible? => true, :admin? => true, :approver? => false, :owner? => false) }
+      before do
+        ENV['AUTO_APPROVAL'] = 'y'
+        allow(RBAC::Access).to receive(:new).with('requests', 'create').and_return(access_obj)
+        allow(access_obj).to receive(:process).and_return(access_obj)
+        post "#{api_version}/workflows/#{workflow_id}/requests", :params => valid_attributes, :headers => request_header
+      end
+
+      after { ENV['AUTO_APPROVAL'] = nil }
+
+      it 'returns status code 201' do
+        expect(response).to have_http_status(201)
+      end
+    end
+
+    context 'admin role when no permission' do
+      let(:access_obj) { instance_double(RBAC::Access, :accessible? => true, :admin? => true, :approver? => false, :owner? => false) }
+      before do
+        allow(RBAC::Access).to receive(:new).with('requests', 'create').and_return(access_obj)
+        allow(access_obj).to receive(:process).and_return(access_obj)
+        post "#{api_version}/workflows/#{workflow_id}/requests", :params => valid_attributes, :headers => request_header
+      end
+
+      it 'returns status code 500' do
+        expect(response).to have_http_status(500)
+      end
+    end
+
+    context 'approver role' do
+      let(:access_obj) { instance_double(RBAC::Access, :accessible? => true, :admin? => false, :approver? => true, :owner? => false) }
+      before do
+        ENV['AUTO_APPROVAL'] = 'y'
```

Comment:
@hsong-rh can we use the with_modified_env instead of setting the ENV variables directly